### PR TITLE
FTX: Add REST subaccount support

### PR DESCRIPTION
**Pull request by thrasher-**
# PR Description

Adds REST API subaccount support. NOTE: There's currently a duplication issue for portfolio info introduced in https://github.com/thrasher-corp/gocryptotrader/pull/530. Will be addressing this in a separate PR. Also fixes a particular spelling mistake in the codebase 😆 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Travis with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/ftx/ftx_test.go [hunk 1]
```diff
@@ -1450,3 +1451,115 @@ func TestParsingWSOBData2(t *testing.T) {
...
+	if !errors.Is(err, errSubaccountNameMustBeSpecified) {
+		t.Errorf("expected %s, but received: %s", errSubaccountNameMustBeSpecified, err)
+	}
+	if !areTestAPIKeysSet() {
+		t.Skip("skipping test, api keys not set")
+	}
+	_, err = f.SubaccountBalances("non-existent")
+	if err == nil {
+		t.Error("expecting non-existent subaccount to return an error")
+	}
```
Comment (gloriousCode) [inline exchanges/ftx/ftx_test.go:1537]: Could you please add a successful scenario here?
```
_, err = f.CreateSubaccount("subzero")
	if err != nil {
		t.Fatal(err)
	}
	_, err = f.SubaccountBalances("subzero")
	if err != nil {
		t.Error(err)
	}
	if err := f.DeleteSubaccountName("subzero"); err != nil {
		t.Error(err)
	}
```

File: exchanges/ftx/ftx.go [hunk 2]
```diff
@@ -1105,3 +1115,107 @@ func (f *FTX) GetOTCQuoteStatus(marketName, quoteID string) ([]QuoteStatusData,
...
+		Data Subaccount `json:"result"`
+	}{}
+	if err := f.SendAuthHTTPRequest(exchange.RestSpot, http.MethodPost, subaccountsUpdateName, d, &resp); err != nil {
+		return nil, err
+	}
+	return &resp.Data, nil
+}
+
+// DeleteSubaccountName deletes the specified subaccount name
+func (f *FTX) DeleteSubaccountName(name string) error {
```
Comment (gloriousCode) [inline exchanges/ftx/ftx.go:1163]: `DeleteSubAccountByName` or `DeleteSubAccount`? Since its actually deleting the account and not the name

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Not a fan of FTX telling me I'm  `"special":false` 👀 
But tested everything and it works, good stuff! Just some minor niterinos

**Review by gloriousCode — APPROVED** [review]
Thanks for making those changes! tACK!

File: exchanges/ftx/ftx_test.go [hunk 3]
```diff
@@ -1450,3 +1451,126 @@ func TestParsingWSOBData2(t *testing.T) {
...
+		ErrExpected error
+	}{
+		{ErrExpected: errCoinMustBeSpecified},
+		{Coin: currency.BTC, ErrExpected: errSubaccountTransferSizeGreaterThanZero},
+		{Coin: currency.BTC, Size: 420, ErrExpected: errSubaccountTransferSourceDestinationMustNotBeEqual},
+	}
+	for x := range tt {
+		_, err := f.SubaccountTransfer(tt[x].Coin, tt[x].Source, tt[x].Destination, tt[x].Size)
+		if !errors.Is(err, tt[x].ErrExpected) {
+			t.Errorf("expected %s, but receieved: %s", tt[x].ErrExpected, err)
```
Comment (MadCozBadd) [inline exchanges/ftx/ftx_test.go:1566]: receieved -> received :D

File: exchanges/ftx/ftx_types.go [hunk 4]
```diff
@@ -234,18 +234,18 @@ type WalletCoinsData struct {
...
-	Coin  string  `json:"coin"`
-	Free  float64 `json:"free"`
-	Total float64 `json:"total"`
+// WalletBalance stores balances data
+type WalletBalance struct {
+	Coin                   string  `json:"coin"`
+	Free                   float64 `json:"free"`
+	Total                  float64 `json:"total"`
+	AvailableWithoutBorrow float64 `json:"availableWithoutBorrow"`
+	USDValue               float64 `json:"usdValue"`
```
Comment (MadCozBadd) [inline exchanges/ftx/ftx_types.go:243]: Can't seem to find the 'usdValue' field in their api docs anymore. Maybe they mightve removed it

**Review by MadCozBadd — COMMENTED** [review]
Just two little things. Everything else is perfect!

**Review by MadCozBadd — COMMENTED** [review]
Also one more thing since you have removed all the receieved spelling errors from other files too, could you remove the spelling mistake from currency/storage_types.go ln 49 please.

**Review by shazbert — CHANGES_REQUESTED** [review]
pedantic nit 😆 all I could find. Everything else is good and tested.

**Review by gloriousCode — COMMENTED** [review]
In the interest in squashing all the receivieiveieveds in our codebase, [ZB has also fixed their API typo](https://www.zb.com/api#suxpnkgwxunfwnv). Fix would be under `zb.go` line 439 😄 

I re-approve of existing changes though 🤙

**Review by MadCozBadd — APPROVED** [review]
tACK!!!!!!!

**Review by shazbert — APPROVED** [review]
tACK!

**Review by gloriousCode — APPROVED** [review]
Thanks for fixing ZB too :D retack

**Pull request merged**